maestro-client/src/main/java/maestro/MaestroTimer.kt

```diff
@@ -23,10 +23,13 @@ object MaestroTimer {
         return null
     }
 
-    fun retryUntilTrue(timeoutMs: Long, block: () -> Boolean): Boolean {
+    fun retryUntilTrue(timeoutMs: Long, delayMs: Long? = null, block: () -> Boolean): Boolean {
         val endTime = System.currentTimeMillis() + timeoutMs
         do {
             try {
+                delayMs?.let {
+                    sleep(Reason.BUFFER, delayMs)
+                }
```

Comment:
Curious how delay is preventing the hang we are facing?

Reply:
it seems that there are two issues:
1. polling `Socket(idbHost, idbPort)` too fast, adding some delay here is healthy, may be decreasing it to 500ms or smth would be fine. same for further operations like `iosDevice.contentDescriptor()`
2. this do->while thing hangs infinitely even after 60s timeout is passed, so I'm not sure if we could rely on this wrapper function long-term at all, I'd rather use RxJava or Coroutines instead here.

as a quickfix I'd suggest to merge this now, because many users can be involved. and then work on more robust approach meanwhile.

Reply:
> as a quickfix I'd suggest to merge this now
Agree, to have this merged today (before the launch event)

Can the delay be inserted after the first run of `block()`?

Reply:
Lets also set the delay to something like 100-250ms. 100ms is a really long pause compared to local-socket validation time.

Reply:
yes, it doesn't timeout while hanging for me, does it for you? 

I think such utility classes for things like retry, retryWithDelay, poll and relying on `System.currentTimeMillis()` is not the best practice, feels like reinventing the wheel a bit, because there will be always some edge cases. In frameworks like RxJava / Coroutines such things are implemented in a more robust way and are better tested.

Reply:
found this one https://github.com/michaelbull/kotlin-retry, which is pretty lightweight, but works on top of coroutines 🤔

Reply:
No objection to adding a dependency to kotlin-retry

Reply:
Tested the branch locally, looks to be stable and fast!